http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -509,16 +507,26 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       hasDeletePermission = listOfPermissions.toSet.contains(AppAction.DeleteApp)
       _ <- if (hasDeletePermission) F.unit else F.raiseError[Unit](ForbiddenError(userInfo.userEmail))
 
-      // Mark the app, nodepool and cluster as deleted in Leo's DB
-      _ <- dbReference.inTransaction(appQuery.markAsDeleted(dbApp.app.id, ctx.now))
-      _ <- dbReference.inTransaction(nodepoolQuery.markAsDeleted(dbApp.nodepool.id, ctx.now))
-      _ <- dbReference.inTransaction(kubernetesClusterQuery.markAsDeleted(dbApp.cluster.id, ctx.now))
-      // Delete kubernetes-app Sam resource
-      _ <- samService.deleteResource(userInfo.accessToken.token, dbApp.app.samResourceId)
-      // Stop the usage of the SAS app
-      _ <- appUsageQuery.recordStop(dbApp.app.id, ctx.now).recoverWith { case e: FailToRecordStoptime =>
-        log.error(ctx.loggingCtx)(e.getMessage)
+      // check if the app is active
+      dbAppActiveOpt <- KubernetesServiceDbQueries
+        .getActiveFullAppByName(cloudContext, appName)
+        .transaction
+      _ <- dbAppActiveOpt match {
+        case Some(dbAppActive) =>
+          for {
```

Comment:
you have already retrieved the `App` at line 490, above. Instead of re-querying the db, can you just match on that app's `status` and look for `AppStatus.Deleted`?

I see that `KubernetesServiceDbQueries.getActiveFullAppByName` does additional work related to clusters/cloud context, so I don't know if these are truly equivalent.

Reply:
The AppRecord has the `id` and `nodepoolId`, but not the cluster id. Instead of `KubernetesServiceDbQueries
        .getActiveFullAppByName(cloudContext, appName)` I could call `kubernetesClusterQuery.getMinimalActiveClusterByCloudContext` to just get the cluster id. Or update the query to get the app record to also include the cluster info.

Reply:
ah, so they are actually different and required - no worries, then!

Reply:
nit, but it might be helpful if you added that context in a comment in this area, I agree with the reasoning though! You could probably add a `getFullAppByName` to `KubernetesServiceDbQueries` to grab everything in one query, but I'm also fine with the current solution